tool/README.md

```diff
@@ -1,130 +1,159 @@
 
 ```shell
 ./tool/update_flutter_sdk.sh
 ``` 
 
-### Prep to publish
+#### Build DevTools for publishing
 
 ```shell
 ./tool/publish.sh
-``` 
+```
 
-### Publish
-#### Verify the package works (DevTools)
+#### Verify the DevTools binary works
 
-- Launch the devtools server
+- Launch the DevTools server
 ```
-cd packages/devtools
-dart bin/devtools.dart
+dart packages/devtools/bin/devtools.dart
 ```
 - open the page in a browser (http://localhost:9100)
-- flutter run an application
-- connect to the running app from devtools, and verify that the pages
-  generally work, and there are no exceptions in the chrome devtools log
+- `flutter run` an application
+- connect to the running app from DevTools, verify that the pages
+  generally work and that there are no exceptions in the chrome devtools log
 
 #### Publish the packages
 
 ```shell
 ./tool/pub_publish.sh
 ```
 
-#### Revert the change to .gitignore and pubspec files
+#### Revert changes caused by the `publish.sh` script
 ```shell
-git checkout .gitignore
-git checkout packages/*/pubspec.yaml
-```
-
-Update the version of DevTools referenced by
-https://github.com/flutter/flutter/blob/8c4538618f81a42e45cdb03c6a204f6e7bcf81b3/dev/devicelab/lib/tasks/perf_tests.dart#L1448
-
-with a CL like
-https://github.com/flutter/flutter/pull/81869
-
-to avoid the issue described by https://github.com/flutter/flutter/issues/81552
-Once DevTools is deployed as part of the Dart SDK we should be able to eliminate this step.
+git checkout .
 
+git clean -f -d
+```
 
 #### Create the tag for this release and push to the remote repository.
 This script will automatically determine the version from the `packages/devtools/pubspec.yaml` so there
 is no need to manually enter the version.
 ```shell
 tool/tag_version.sh
 ```
+
+#### Upload the DevTools binary to CIPD
+
+Checkout the version of DevTools you just tagged (for example, `2.7.0`):
+```shell
+git checkout v2.7.0
+
+Note: switching to 'v2.7.0'.
+
+You are in 'detached HEAD' state. You can look around, make experimental
+changes and commit them, and you can discard any commits you make in this
+state without impacting any branches by switching back to a branch.
+
+If you want to create a new branch to retain commits you create, you may
+do so (now or later) by using -c with the switch command. Example:
+
+  git switch -c <new-branch-name>
+
+Or undo this operation with:
+
+  git switch -
+
+Turn off this advice by setting config variable advice.detachedHead to false
+
+HEAD is now at 8881a7ca Update release scripts (#3426)
+```
+
+Copy the commit hash from the bottom of the CLI output (in the example above, `8881a7ca`).
+
+Using the [update.sh](https://github.com/dart-lang/sdk/blob/master/third_party/devtools/update.sh)
+script at https://github.com/dart-lang/sdk/tree/master/third_party/devtools, build DevTools at the 
+given git commit hash, and upload the binary to CIPD.
+
+```shell
+sdk/third_party/devtools/update.sh 8881a7ca
```

Comment:
This command actually pulls down everything you need to build and publish to CIPD, so there's no need to be in a particular branch or to build DevTools before running this script, and it won't nuke any local changes in your DevTools repo. The temporary copy will be deleted once the script is complete.

Reply:
Gotcha. Since this is the last pub release (though we may have one more that gives a note about `dart devtools`, I'll remove all the pub related instructions.